Add :abort_if_pending_migrations rake task

tomgi opened the pull request:
Added 
```
rake data:abort_if_pending_migrations
rake db:abort_if_pending_migrations:with_data
```
as an addition / extension to
```
rake db:abort_if_pending_migrations
```
provided by rails.

The new tasks could be useful during deployment process to ensure that all data migrations have been executed before starting the servers.

In the project that I'm currently working on, we use `rake db:abort_if_pending_migrations` during deployment pipeline to ensure that ActiveRecord migrations have been executed before starting the servers. We would like to extend that to data migrations as well.

ilyakatz commented on tasks/databases.rake:
```diff
@@ -367,6 +381,16 @@ def sort_string migration
 end
 
+def abort_if_pending_migrations(migrations, message)
+  if migrations.any?
```
this rake file has been neglected, so for any new changes, i've been trying not to add more lines to the file. hence, i started putting logic for rake taks into `lib/data_migrate/tasks/data_migrate_tasks.rb`. would you mind adding this in there

tomgi replied:
Sure, done

ilyakatz commented:
thanks for this PR. I added a comment. also, do you know why rails 6 is failing? wonder if the latest changes fix this

tomgi commented:
Thanks @ilyakatz, I merged the new master into this branch - rails 6 tests are now passing.

tomgi commented:
Hey @ilyakatz, do you think we could merge this PR?

ilyakatz commented:
Hi @tomgi. Sorry I forgot about this PR. I just tried to create a branch that has master rebased and I see there are failed tests. (https://github.com/ilyakatz/data-migrate/pull/146). Do you know why those are failing? Will be happy to merge and deploy once tests are passing

tomgi commented:
HI @ilyakatz - I rebased and squashed this PR on your latest master and force-pushed it.
All the tests are now passing :)

 I think the tests are failing on your PR because it seems to be missing some changed from master - e.g. https://github.com/ilyakatz/data-migrate/pull/146/files#diff-235b1e857df6b90c9ee64688ab65b813R14-R32 has been removed from master - possibly a conflict resolution issue?

ilyakatz commented:
@tomgi merged and deployed to rubygems
